src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -325,15 +325,32 @@ private void LogPrimaryKeys()
             {
                 // row["DATA_TYPE"] has value type string so a direct cast to System.Type is not supported.
                 // See https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/sql-server-data-type-mappings
-                Type systemType = SqlToCLRType((string)row["DATA_TYPE"]);
-                // Add to parameters dictionary without the leading @ sign
-                storedProcedureDefinition.Parameters.TryAdd(((string)row["PARAMETER_NAME"])[1..],
-                    new()
+                string sqlType = (string)row["DATA_TYPE"];
+                Type systemType = SqlToCLRType(sqlType);
+                ParameterDefinition paramDefinition = new()
+                {
+                    SystemType = systemType,
+                    DbType = TypeHelper.GetDbTypeFromSystemType(systemType)
+                };
+
+                if (GetDatabaseType() is DatabaseType.MSSQL &&
```

Comment:
This is a base class implementation of `FillSchemaForStoredProcedureAsync`. If there is certain portion of the code that really is only meant for a particular type of db, I think this function should get a overridden definition `FillSchemaForStoredProcedureAsync`. 
The parts of code that are common for all dbs can still live in the base class - so the overridden function calls the parts that are common and does additional stuff where needed. Thats why we have the derived classes.

Reply:
Done.